Tools/clinic/clinic.py

```diff
@@ -850,6 +866,56 @@ def render(
                 function = o
         return self.render_function(clinic, function)
 
+    def deprecate_positional_use(
+            self,
+            func: Function,
+            params: dict[int, Parameter],
+    ) -> str:
+        assert len(params) > 0
+        # FIXME: Handle multiple deprecation levels
+        # FIXME: For now, assume there's only one level
+        code_blocks: list[str] = []
+        names = [repr(p.name) for p in iter(params.values())]
```

Comment:
Useless use of `iter()`: `params.values()` is already iterable:

```suggestion
        names = [repr(p.name) for p in params.values()]
```

Reply:
It is not an iterator.

Reply:
But it is an iterable

Reply:
"Iterable" = The class of the object defines `__iter__`. You can directly iterate over any iterable, and you can call `iter()` on any iterable. But you can't call `next()` on any iterable unless the iterable is also an iterator.

"Iterator" = The class of the object defines `__next__` (and in 99.9% of cases, `__iter__` as well). You can call `next()` on any iterator.

Reply:
```python
>>> next({"a": 1}.values())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_values' object is not an iterator
```

Reply:
Oops, you are of course right. I'll fix it; thanks!

Reply:
Yes, but you're not calling `next()` here, you're just iterating over it. See my comment immediately above. You don't need an iterator, you just need an iterable:

```pycon
>>> list({'a': 1}.values())
[1]
```